blockchain/blockdao/blockdao.go

```diff
@@ -37,8 +38,14 @@ var (
 type (
 	// BlockDAO represents the block data access object
 	BlockDAO interface {
-		Start(ctx context.Context) error
-		Stop(ctx context.Context) error
+		BlockStore
+		GetBlob(hash.Hash256) (*types.BlobTxSidecar, string, error)
```

Comment:
Do we really need to the two methods `GetBlob` and `GetBlobsByHeight`? We may can simply fill it when returning the block if there is a blob